src/libraries/System.Private.CoreLib/src/System/Globalization/IcuLocaleData.cs

```diff
@@ -4510,70 +3839,64 @@ internal static string GetConsoleUICulture(string cultureName)
             return GetLocaleDataMappedCulture(cultureName, IcuLocaleDataParts.ConsoleLocaleIndex);
         }
 
-        // SearchCultureName will binary search c_localeNames using s_localeNamesIndices.
-        // return index in s_localeNamesIndices, or -1 if it fail finding any match
+        // Returns index of the culture or -1 if it fail finding any match
         private static int SearchCultureName(string name)
         {
-            int left = 0;
-            int right = s_localeNamesIndices.Length - 2;
-            int index;
-            int result;
+            if (name.Length > LocaleLongestName)
+                return -1;
 
-            Debug.Assert(s_localeNamesIndices[s_localeNamesIndices.Length - 1] == c_localeNames.Length);
+            Span<byte> lower_case = stackalloc byte[name.Length];
```

Comment:
Could the name that comes in here ever be provided by a caller, such that (in an erroneous case) this could crash with a stack overflow?

Reply:
No, there is check for length 2 lines above https://github.com/dotnet/runtime/pull/45296/files#diff-423b39c173efa91a947f7f39560ae5aa07a690040b4cc6438240107df45bdda9R3845